include/mxnet/mxruntime.h

```diff
@@ -0,0 +1,194 @@
+        /*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file mxruntime.h
+ * \brief check MXNet features including compile time support
+ */
+
+#pragma once
+
+#include <string>
+#include "dmlc/base.h"
+#include "mshadow/base.h"
+
+/*!
+ *\brief whether to use opencv support
+ */
+#ifndef MXNET_USE_OPENCV
+#define MXNET_USE_OPENCV 1
+#endif
+
+/*!
+ *\brief whether to use cuda support
+ */
+#ifndef MXNET_USE_CUDA
+#define MXNET_USE_CUDA MSHADOW_USE_CUDA
+#endif
+
+/*!
+ *\brief whether to use cudnn library for convolution
+ */
+#ifndef MXNET_USE_CUDNN
+#define MXNET_USE_CUDNN MSHADOW_USE_CUDNN
+#endif
+
+#ifndef MXNET_USE_NCCL
+#define MXNET_USE_NCCL 0
+#endif
+
+/*!
+ *\brief whether to use cusolver library
+ */
+#ifndef MXNET_USE_CUSOLVER
+#define MXNET_USE_CUSOLVER MSHADOW_USE_CUSOLVER
+#endif
+
+#ifndef MXNET_ENABLE_CUDA_RTC
+#define MXNET_ENABLE_CUDA_RTC 0
+#endif
+
+/*! \brief Error message for using gpu when MXNET_USE_CUDA==0 */
+#define MXNET_GPU_NOT_ENABLED_ERROR  "GPU is not enabled"
+
+
+#ifndef MXNET_USE_TENSORRT
+#define MXNET_USE_TENSORRT 0
+#endif
+
+
+#ifndef MXNET_USE_BLAS_ATLAS
+#define MXNET_USE_BLAS_ATLAS 0
+#endif
+
+#ifndef MXNET_USE_BLAS_OPEN
+#define MXNET_USE_BLAS_OPEN 0
+#endif
+
+#ifndef MXNET_USE_BLAS_MKL
+#define MXNET_USE_BLAS_MKL 0
+#endif
+
+#ifndef MXNET_USE_BLAS_APPLE
+#define MXNET_USE_BLAS_APPLE 0
+#endif
+
+#ifndef MXNET_USE_LAPACK
+#define MXNET_USE_LAPACK 0
+#endif
+
+#ifndef MXNET_USE_MKLDNN
+#define MXNET_USE_MKLDNN 0
+#endif
+
+#ifndef MXNET_USE_OPENMP
+#define MXNET_USE_OPENMP 0
+#endif
+
+#ifndef MXNET_USE_F16C
+#define MXNET_USE_F16C MSHADOW_USE_F16C
+#endif
+
+#ifndef MXNET_USE_CAFFE
+#define MXNET_USE_CAFFE 0
+#endif
+
+#ifndef MXNET_USE_DIST_KVSTORE
+#define MXNET_USE_DIST_KVSTORE 0
+#endif
+
+#ifndef MXNET_USE_SIGNAL_HANDLER
+#define MXNET_USE_SIGNAL_HANDLER 0
+#endif
+
+
+
+namespace mxnet {
+namespace features {
+// Check compile flags such as CMakeLists.txt
+
+/// Compile time features
+// ATTENTION: When changing this enum, match the strings in mxruntime.cc!!!
```

Comment:
Could we have a sanity check script that ensures that they are in sync? It could then throw an error if they are out of sync.

Reply:
Or is that what tests/cpp/misc/runtime_test.cc is supposed to do?

Reply:
Yes the test checks that they have the same number of elements, I think is enough to prevent them going out of sync. I think is enough.